File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    
    queue = deque()
    visit = [False] * n
    
    for i in range(n) :
        if visit[i] :
            continue
        
        queue.append(i)
        visit[i] = True
        answer += 1
        
        while queue :
            current_computer = queue.popleft()
            
            for j in range(n) :
                if computers[current_computer][j] == 1 and not visit[j] :
                    visit[j] = True
                    queue.append(j)
            
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visit = new boolean[n]; // false
        // for (int i=0; i<n; i++) {
        //     System.out.print(visit[i]);
        //     System.out.print(" ");
        // }
        
        Queue<Integer> queue = new LinkedList<>();
        // queue.offer(0);
        // queue.peek();
        // queue.poll();
        // System.out.println(queue.isEmpty());
        for (int i=0; i<n; i++) {
            if (visit[i]) {
                continue;
            }
            // System.out.println("------");
            // System.out.println("i:" + i);
            visit[i] = true;
            queue.offer(i);
            answer++;
            
            while (!queue.isEmpty()) {
                 int value = queue.poll();
                
                for (int j=0; j<n; j++) {
                    if (computers[value][j] == 1 && !visit[j]) {
                        visit[j] = true;
                        queue.offer(j);
                    }
                }
            }
        }
        
        return answer;
    }
}